simplecpp.cpp

```diff
@@ -1843,67 +1843,91 @@ namespace simplecpp {
 class ScopedLock
 {
 public:
-    explicit ScopedLock(HANDLE mutex)
-        : m_mutex(mutex)
+    explicit ScopedLock(CRITICAL_SECTION& criticalSection)
+        : m_criticalSection(criticalSection)
     {
-        if (WaitForSingleObject(m_mutex, INFINITE) == WAIT_FAILED)
-            throw std::runtime_error("cannot lock the mutex");
+        EnterCriticalSection(&m_criticalSection);
     }
 
     ~ScopedLock()
     {
-        ReleaseMutex(m_mutex);
+        LeaveCriticalSection(&m_criticalSection);
     }
 
 private:
     ScopedLock& operator=(const ScopedLock&);
     ScopedLock(const ScopedLock&);
 
-    HANDLE m_mutex;
+    CRITICAL_SECTION& m_criticalSection;
 };
 
-class RealFileNameMap
+template <typename TKey, typename TValue> class ThreadSafeCache
```

Comment:
I am not sure, you generalize the code but it seems the original class name and method names were more explicit... yes this could be useful in the future for some other purpose. But right now it is a file name map, the key type is a string and the value type is string. It is overkill with a template here.

Your changes are not wrong but I suggest that the original class name and method names are restored.